test/integration/controller/admissionchecks/provisioning/provisioning_test.go

```diff
@@ -568,39 +569,69 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
-			createdRequest := &autoscaling.ProvisioningRequest{}
 			provReqKey := types.NamespacedName{
 				Namespace: wlKey.Namespace,
 				Name:      provisioning.GetProvisioningRequestName(wlKey.Name, ac.Name, 1),
 			}
-
 			ginkgo.By("Checking that the provision request is created", func() {
+				createdRequest := &autoscaling.ProvisioningRequest{}
 				gomega.Eventually(func() error {
 					return k8sClient.Get(ctx, provReqKey, createdRequest)
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
 			ginkgo.By("Setting the provision request as Provisioned", func() {
-				gomega.Eventually(func() error {
+				createdRequest := &autoscaling.ProvisioningRequest{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, provReqKey, createdRequest)).To(gomega.Succeed())
 					apimeta.SetStatusCondition(&createdRequest.Status.Conditions, metav1.Condition{
 						Type:   autoscaling.Provisioned,
 						Status: metav1.ConditionTrue,
 						Reason: autoscaling.Provisioned,
 					})
-					return k8sClient.Status().Update(ctx, createdRequest)
+					g.Expect(k8sClient.Status().Update(ctx, createdRequest)).To(gomega.Succeed())
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
 			ginkgo.By("Checking the admission check is ready", func() {
+				updatedWl := &kueue.Workload{}
 				gomega.Eventually(func(g gomega.Gomega) {
 					g.Expect(k8sClient.Get(ctx, wlKey, updatedWl)).To(gomega.Succeed())
 					state := workload.FindAdmissionCheck(updatedWl.Status.AdmissionChecks, ac.Name)
 					g.Expect(state).NotTo(gomega.BeNil())
 					g.Expect(state.State).To(gomega.Equal(kueue.CheckStateReady))
+					g.Expect(state.PodSetUpdates).To(gomega.BeComparableTo([]kueue.PodSetUpdate{
+						{
+							Name: "ps1",
+							Annotations: map[string]string{
+								provisioning.ConsumesAnnotationKey: provReqKey.Name,
+							},
+						},
+						{
+							Name: "ps2",
+							Annotations: map[string]string{
+								provisioning.ConsumesAnnotationKey: provReqKey.Name,
+							},
+						},
+					}))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Check the workload is admitted", func() {
+				updatedWl := &kueue.Workload{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, updatedWl)).To(gomega.Succeed())
+					util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, string(updatedWl.Status.Admission.ClusterQueue), updatedWl)
```

Comment:
Should we use `ExpectWorkloadsToBeAdmitted` instead of `ExpectWorkloadsToHaveQuotaReservation`?

https://github.com/kubernetes-sigs/kueue/blob/395137afeaad94179db68043e5c74ab193f6beb1/test/util/util.go#L268

Since the provisioning controller verifies `Admission` status:

https://github.com/kubernetes-sigs/kueue/blob/395137afeaad94179db68043e5c74ab193f6beb1/pkg/controller/admissionchecks/provisioning/controller.go#L138-L142

https://github.com/kubernetes-sigs/kueue/blob/395137afeaad94179db68043e5c74ab193f6beb1/pkg/workload/workload.go#L433-L436

Reply:
Sounds resonable to me.

Reply:
yes, I think we should use `ExpectWorkloadsToBeAdmitted` to meet the `IsAdmitted` check. 

As discussed https://github.com/kubernetes-sigs/kueue/pull/1493#issuecomment-1865853179, I suspect this is problematic because there is no scheduler in the integration test setup. Please consider calling `SyncAdmittedConditionForWorkloads`.